### PR TITLE
FSET-1758: Ensure accepted FastPass candidates are progressed to SIFT

### PR DESCRIPTION
**Pull request by fayimora**

### CODE REVIEW

**Comment by henricook** [comment]
@fayimora what happens if the schemes they've selected aren't siftable? (e.g. just choosing human resources)?

**Comment by fayimora** [comment]
@henricook Good catch! My guess is they should get moved straight to `FSAC_AWAITING_ALLOCATION`. I'll confirm with Andy so he can update the ticket.

**Comment by henricook** [comment]
:+1: - i will DNM temporarily

File: app/services/fastpass/FastPassService.scala [hunk 1]
```diff
@@ -82,6 +87,19 @@ trait FastPassService extends EventSink {
...
 
+  private def autoProgressToSiftOrFSAC(applicationId: String): Future[Unit] = {
+    for {
+      preferences <- schemePreferencesService.find(applicationId)
+    } yield {
+      val hasSiftableScheme = schemesRepository.siftableSchemeIds.intersect(preferences.schemes).isEmpty
+      if(hasSiftableScheme) {
+        appRepo.addProgressStatusAndUpdateAppStatus(applicationId, ProgressStatuses.SIFT_ENTERED)
+      } else {
+        appRepo.addProgressStatusAndUpdateAppStatus(applicationId, ProgressStatuses.ASSESSMENT_CENTRE_AWAITING_ALLOCATION)
```
Comment (henricook) [inline app/services/fastpass/FastPassService.scala]: I suspect this is returning a Future[Future[Unit]] - which means you'll have an unmapped future without a .flatMap(identity)

Comment (henricook) [reply app/services/fastpass/FastPassService.scala]: The way Future[Unit] eats other futures makes me so sad :-(

Comment (fayimora) [reply app/services/fastpass/FastPassService.scala]: This is very strange behaviour. I'm very curious why this happens(perhaps some weird type erasure). Thanks for the catch

**Pull request merged**